app/classifier/tasks/text/index.jsx

```diff
@@ -0,0 +1,189 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash';
+
+import GenericTask from '../generic';
+import TextTaskEditor from './editor';
+import TextTaskSummary from './summary';
+
+const LINEHEIGHT = 22.5;
+const NOOP = Function.prototype;
+
+export default class TextTask extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rows: 1,
+      value: ''
+    };
+
+    this.textInput = React.createRef();
+
+    this.debouncedUpdateAnnotation = _.debounce(this.updateAnnotation, 500).bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+    this.setTagSelection = this.setTagSelection.bind(this);
+    this.setValue = this.setValue.bind(this);
+    this.updateAnnotation = this.updateAnnotation.bind(this);
+  }
+
+  componentDidMount() {
+    this.setValue();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.task && this.props.task && (prevProps.task !== this.props.task)) {
+      const value = prevState.value;
+      const newAnnotation = Object.assign(prevProps.annotation, { value });
+      prevProps.onChange(newAnnotation);
```

Comment:
Lines 37-39 and on line 119 the `onBlur={this.updateAnnotation}` are from when I was trying to avoid any updateAnnotation in `handleChange`, but `handleChange` needs some updateAnnotation so the container knows the task has been completed, otherwise I ran into the scenario with a required text task where I would have typed an answer, but not triggered the `onBlur` and it would think I had not typed anything and would still have "Next" or "Done" disabled when it shouldn't have been.

With `onBlur={this.updateAnnotation}` lines 37-39 may not be needed?

I've kept `onBlur={this.updateAnnotation}` for the scenario of back-to-back two text task combos. If the first text task is completed, then the second, then next task, neither text task is unmounted and with the debounced updateAnnotation I'm not sure if there's a chance the last few keypresses could be missed, but the `onBlur` would make sure those would be saved?